src/core/server/plugins/types.ts

```diff
@@ -24,7 +24,49 @@ import { ConfigPath, EnvironmentMode, PackageInfo } from '../config';
 import { LoggerFactory } from '../logging';
 import { CoreSetup, CoreStart } from '..';
 
-export type PluginConfigSchema = Type<unknown> | null;
+/**
+ * Dedicated type for plugin configuration schema.
+ *
+ * @public
+ */
+export type PluginConfigSchema<T = unknown> = Type<T>;
+
+/**
+ * Describes a plugin configuration schema and capabilities.
+ *
+ * @example
+ * ```typescript
+ * // my_plugin/server/index.ts
+ * import { schema, TypeOf } from '@kbn/config-schema';
+ * import { PluginConfigDescriptor } from 'kibana/server';
+ *
+ * const configSchema = schema.object({
+ *   secret: schema.string({ defaultValue: 'Only on server' }),
+ *   uiProp: schema.string({ defaultValue: 'Accessible from client' }),
+ * });
+ *
+ * type ConfigType = TypeOf<typeof configSchema>;
+ *
+ * export const config: PluginConfigDescriptor<ConfigType> = {
+ *   exposeToBrowser: ['uiProp'],
+ *   schema: configSchema,
+ * };
+ * ```
+ *
+ * @public
+ */
+export interface PluginConfigDescriptor<T = any> {
+  /**
+   * List of configuration properties that will be available on the client-side plugin.
+   */
+  exposeToBrowser?: Array<keyof T>;
```

Comment:
Though we're only supporting top-level keys at this time, I think it would be more future proof if we type this as an object of string-boolean pairs. This could be extended in the future to support nested keys.

```ts
exposeToBrowser?: { [P in keyof T]?: boolean };
```

Plugins would then specify this like:
```ts
export const config: PluginConfigDescriptor<ConfigType> = {
  exposeToBrowser: {
    uiProp: true,
  },
  schema: configSchema,
};
```

Reply:
To clarify, I think `{ [P in keyof T]?: boolean }` is a better type to use now over `Array<keyof T>` because it is compatible with a type that we could use in the future for nested keys:

```ts
type BooleanValues<T> = {
  [P in keyof T]?: T[P] extends object ? boolean | BooleanValues<T> : boolean
}
```

By using a compatible type now, we can easily add nested key support in the future without breaking all existing `exposeToBrowser` configurations.